hbase-client/src/main/java/org/apache/hadoop/hbase/client/replication/ReplicationPeerConfigUtil.java

```diff
@@ -64,6 +65,9 @@
   public static final String HBASE_REPLICATION_PEER_BASE_CONFIG =
     "hbase.replication.peer.base.config";
 
+  public static final String HBASE_REPLICATION_PEER_REMOVE_BASE_CONFIG =
```

Comment:
I think this approach can be better, it seems confusing that one would add a configuration to remove another configuration?  (and the community is generally against adding such esoteric configs)

What happens if someone forgets to undo this and that has some adverse affect after a next restart..

Why not just consider "" config as a special case and then reset the configs in the copiedConfigBuilder?

Something like

initially someone set it k1=v1;k2=v2.... 
then they can set it to "" and restart master

detect that "" has been set (which is different from what we already have) and remove whatever we already have populated?

To avoid config divergence, we can disallow hbase.replication.peer.base.config to be done via updateConfig RPC (throw some sensible exception)..

Reply:
@bharathv do you mean setting k1="" if we want to remove k1 configuration?
Just setting "" doesn't really communicate the program what we want to remove because there might be other other configs added from outside of hbase-site.